reference.md

```diff
@@ -0,0 +1,1393 @@
+# Snapcraft CLI reference
+
+## lifecycle and misc
+
+### version
+
+    $ snapcraft --version
+    2.12.1
+
+### help
+
+    $ snapcraft help
+    Usage:
+      snapcraft [options] [--enable-geoip --no-parallel-build]
+      snapcraft [options] init
+      snapcraft [options] pull [<part> ...]  [--enable-geoip]
+      snapcraft [options] build [<part> ...] [--no-parallel-build]
+      snapcraft [options] stage [<part> ...]
+      snapcraft [options] prime [<part> ...]
+      snapcraft [options] strip [<part> ...]
+      snapcraft [options] clean [<part> ...] [--step <step>]
+      snapcraft [options] snap [<directory> --output <snap-file>]
+      snapcraft [options] cleanbuild
+      snapcraft [options] login
+      snapcraft [options] logout
+      snapcraft [options] register <snap-name>
+      snapcraft [options] upload <snap-file>
+      snapcraft [options] push <snap-file> [--release <channels>]
+      snapcraft [options] release <snap-name> <revision> <channel>
+      snapcraft [options] list-plugins
+      snapcraft [options] tour [<directory>]
+      snapcraft [options] update
+      snapcraft [options] define <part-name>
+      snapcraft [options] search [<query> ...]
+      snapcraft [options] help (topics | <plugin> | <topic>) [--devel]
+      snapcraft (-h | --help)
+
+
+    $ snapcraft --help
+    $ snapcraft -h
+    snapcraft
```

Comment:
This is the description of the command. We need to say something more useful than snapcraft here.